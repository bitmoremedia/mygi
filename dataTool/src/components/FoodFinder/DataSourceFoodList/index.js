import React, { PropTypes } from 'react'
import ToolTip from '../../../components/common/ToolTip'

const DataSourceFoodList = ({ list }) => {
  const renderItem = food =>
    <li key={food.id}>
      {food.name} [{food.gi}]
      <ToolTip
        text={`${food.source} / ${food.id}`}
        position={'left'}
      >
        ?
      </ToolTip>
    </li>

  return (
    <div>
      <ul>
        {list.map(food => renderItem(food))}
      </ul>
    </div>
  )
}

DataSourceFoodList.propTypes = {
  list: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      gi: PropTypes.any.isRequired,
      source: PropTypes.string.isRequired,
    }),
  ).isRequired,
}

export default DataSourceFoodList
