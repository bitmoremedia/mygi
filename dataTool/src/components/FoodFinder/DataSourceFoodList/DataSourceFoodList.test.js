import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import DataSourceFoodList from './index'

const foodList = [
  {
    id: 'abc-123',
    name: 'Kidney Bean',
    gi: 30,
    source: 'havard',
  },
  {
    id: 'def-456',
    name: 'Kidney Bean',
    gi: 44,
    source: 'mendosa',
  },
]

test('component renders without error', () => {
  const wrapper = mount(
    <DataSourceFoodList list={foodList} />,
  )
  expect(wrapper.exists()).toBe(true)
})

// SNAPSHOT TESTS

test('component renders as expected', () => {
  const wrapper = mount(
    <DataSourceFoodList list={foodList} />,
  )
  expect(toJson(wrapper)).toMatchSnapshot()
})
