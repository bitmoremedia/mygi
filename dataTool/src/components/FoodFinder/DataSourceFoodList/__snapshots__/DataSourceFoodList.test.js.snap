exports[`test component renders as expected 1`] = `
<Component
  list={
    Array [
      Object {
        "gi": 30,
        "id": "abc-123",
        "name": "Kidney Bean",
        "source": "havard",
      },
      Object {
        "gi": 44,
        "id": "def-456",
        "name": "Kidney Bean",
        "source": "mendosa",
      },
    ]
  }>
  <div>
    <ul>
      <li>
        Kidney Bean
         [
        30
        ]
        <Component
          length="large"
          position="left"
          text="havard / abc-123">
          <styled.span>
            <span
              className="kpQwSz">
              <span
                data-balloon="havard / abc-123"
                data-balloon-length="large"
                data-balloon-pos="left">
                ?
              </span>
            </span>
          </styled.span>
        </Component>
      </li>
      <li>
        Kidney Bean
         [
        44
        ]
        <Component
          length="large"
          position="left"
          text="mendosa / def-456">
          <styled.span>
            <span
              className="kpQwSz">
              <span
                data-balloon="mendosa / def-456"
                data-balloon-length="large"
                data-balloon-pos="left">
                ?
              </span>
            </span>
          </styled.span>
        </Component>
      </li>
    </ul>
  </div>
</Component>
`;
